core/Test/Tasty/Options/Core.hs

```diff
@@ -102,4 +112,5 @@ coreOptions :: [OptionDescription]
 coreOptions =
   [ Option (Proxy :: Proxy TestPattern)
   , Option (Proxy :: Proxy Timeout)
+  , Option (Proxy :: Proxy HideProgress)
```

Comment:
This option is relevant only to `consoleReporter` ingredient, other ingredients do not care about it. Please move it to `Test.Tasty.Ingredients.ConsoleReporter`.

Reply:
`yieldProgress` is implemented in `Test.Tasty.Run`, it doesn't seem to be just a console reporter thing.